### PR TITLE
[EventHubs] Warning partition key type

### PR DESCRIPTION
**Pull request by yunhaoling**
addressing issue: https://github.com/Azure/azure-sdk-for-python/issues/12353

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py [hunk 1]
```diff
@@ -222,6 +222,8 @@ def send_batch(self, event_data_batch, **kwargs):
          A `TypeError` will be raised if partition_key is specified and event_data_batch is an `EventDataBatch` because
          `EventDataBatch` itself has partition_key.
          If both partition_id and partition_key are provided, the partition_id will take precedence.
+         **WARNING: Please DO NOT pass the partition_key of non-string type. The Event Hub service ignores partition_key
+         of non-string type in which case events will be assigned to all partitions using round-robin.**
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:226]:
```suggestion
         **WARNING: Please DO NOT pass a partition_key of non-string type. The Event Hub service ignores partition_key
         of non-string type, in which case events will be assigned to all partitions using round-robin.**
```

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py [hunk 2]
```diff
@@ -277,6 +279,8 @@ def create_batch(self, **kwargs):
         :keyword str partition_key: With the given partition_key, event data will be sent to
          a particular partition of the Event Hub decided by the service.
          If both partition_id and partition_key are provided, the partition_id will take precedence.
+         **WARNING: Please DO NOT pass the partition_key of non-string type. The Event Hub service ignores partition_key
+         of non-string type in which case events will be assigned to all partitions using round-robin.**
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:283]: same as above

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py (hunk 1 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py]: thanks! just updated!

**Review by swathipil — APPROVED** [review]
lgtm

**Pull request merged**